scio-bigquery/src/test/scala/com/spotify/scio/bigquery/validation/SampleOverrideTypeProvider.scala

```diff
@@ -97,4 +99,24 @@ class SampleOverrideTypeProvider extends OverrideTypeProvider {
   def initializeToTable(c: blackbox.Context)(modifiers: c.universe.Modifiers,
                                              variableName: c.universe.TermName,
                                              tpe: c.universe.Tree): Unit = Unit
+
+
+}
+
+/** This shouldn't be necessary in most production use cases
+  * however passing System properties from Intellij can cause issues
+  *
+  * The ideal place to set this System property is in your build.sbt file
+  */
+object SampleOverrideTypeProvider {
+
+  class setProperty extends StaticAnnotation {
+    def macroTransform(annottees: Any*): Any = macro setPropertyImpl
+  }
+
+  def setPropertyImpl(c: blackbox.Context)(annottees: c.Expr[Any]*): c.Expr[Any] = {
+    System.setProperty("override.type.provider",
```

Comment:
@jbigred1 can we move this to a method? coz then we can probably use it here? https://github.com/spotify/scio/pull/1186/files#diff-81499171e0beaee5a8484487e4abe939R31